packages/forge/src/governance/extensions/GovernorSorting.sol

```diff
@@ -2,232 +2,189 @@
+    // TODO: make RANK_LIMIT configurable + test ranges
+    // TODO: flesh out tests a ton for different edge cases w oldValue, smallestNonZeroSortedRanksValueIdx, and RANK_LIMIT overlaps
+    uint256 public constant RANK_LIMIT = 250; // cannot be 0
+
+    // RULE: array length can never end lower than it started a transaction, otherwise erroneous ranking can happen
+    uint256[] public sortedRanks = new uint256[](RANK_LIMIT); // value is forVotes counts, has the constraint of no duplicate values except for 0 bc it's instantiated with 0 values.
+    uint256 public smallestNonZeroSortedRanksValueIdx = 0; // the index of the smallest non-zero value in sortedRanks, useful to finding where sortedRanks has been populated to
 
     /**
-     * @dev Getter if a given ranking is tied.
+     * @dev Get the number of proposals that have `forVotes` number of for votes.
      */
-    function isTied(uint256 ranking) public view returns (bool) {
-        require(
-            setSortedAndTiedProposalsHasBeenRun, "RewardsModule: run setSortedAndTiedProposals() to populate this value"
-        );
-        return _isTied[ranking];
-    }
+    function getNumProposalsWithThisManyForVotes(uint256 forVotes) public view virtual returns (uint256 count);
 
     /**
-     * @dev Getter for tiedAdjustedRankingPosition of a ranking.
+     * @dev Get the sortedRanks array.
      */
-    function rankingPosition(uint256 ranking) public view returns (uint256) {
-        require(
-            setSortedAndTiedProposalsHasBeenRun, "RewardsModule: run setSortedAndTiedProposals() to populate this value"
-        );
-        return tiedAdjustedRankingPosition[ranking];
+    function getSortedRanks() public view virtual returns (uint256[] memory sortedRanksArray) {
+        return sortedRanks;
     }
 
-    /**
-     * @dev Getter for _sortedProposalIds.
-     */
-    function sortedProposalIds() public view returns (uint256[] memory) {
-        require(
-            setSortedAndTiedProposalsHasBeenRun, "RewardsModule: run setSortedAndTiedProposals() to populate this value"
-        );
-        return _sortedProposalIds;
-    }
+    // get the idx of sortedRanks considered to hold the queried rank taking deleted proposals into account.
+    // a rank has to have > 0 votes to be considered valid.
+    function getRankIndex(uint256 rank) public view returns (uint256 rankIndex) {
+        require(rank != 0, "GovernorSorting: rank cannot equal 0");
 
-    /**
-     * @dev Getter for the lowest ranking.
-     */
-    function lowestRanking() public view returns (uint256) {
-        require(
-            setSortedAndTiedProposalsHasBeenRun, "RewardsModule: run setSortedAndTiedProposals() to populate this value"
-        );
-        return _lowestRanking;
-    }
+        uint256 smallestIdxMemVar = smallestNonZeroSortedRanksValueIdx; // only check state var once to save on gas
+        uint256[] memory sortedRanksMemVar = sortedRanks; // only check state var once to save on gas
```

Comment:
Oo yeah thank you for checking! Sweet, yeah we got that feedback in our audit that it would bc you decrease the number of times you're calling state vars, but good to have confirm!